azurerm_linux_virtual_machine/main.tf

```diff
@@ -29,10 +30,21 @@ resource "azurerm_network_security_group" "network_security_group" {
   }
 
   tags = {
-    environment = "Production"
+    environment = var.environment
   }
 }
 
+resource "azurerm_public_ip" "public_ip" {
+    name                         = local.public_ip_name
```

Comment:
We already have a module fort the public_ip. Use the module.

Reply:
Unfortunately I can't use that module since I used 
```
provider "azurerm" {
  version = "=2.0.0"
.....
```
While there is a constraint to use version 1.44.00 in public ip module.
I can upgrade that module too.

Reply:
It is ok for now, i am going to merge the PR.